recipe/meta.yaml

```
@@ -6,52 +6,40 @@ package:
   version: {{ version }}
 
 source:
-  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz
-  sha256: a59fc3b4fcdb972d09138cb516dbde600c99d07c38fd9372a6ef500d2d031b4e
+  # Use GH to include upstream tests
+  url: https://github.com/jcrist/{{ name }}/archive/refs/tags/{{ version }}.tar.gz
+  sha256: 5bff8c3f9303eb589516e02de4141ce31be09a0fdf4980734b020b64c120295e
 
 build:
   number: 0
-
-outputs:
-  - name: msgspec
-    version: {{ version }}
-    build:
-      skip: true  # [py<38]
-      script: python -m pip install . -vv
-    requirements:
-      build:
-        - python                                 # [build_platform != target_platform]
-        - cross-python_{{ target_platform }}     # [build_platform != target_platform]
-        - {{ compiler('c') }}
-      host:
-        - python
-        - pip
-      run:
-        - python
-    test:
-      imports:
-        - msgspec
-
-  - name: msgspec-yaml
-    version: {{ version }}
-    requirements:
-      run:
-        - {{ pin_subpackage(name) }}
-        - pyyaml
-    test:
-      imports:
-        - msgspec
-
-  - name: msgspec-toml
-    version: {{ version }}
-    requirements:
-      run:
-        - {{ pin_subpackage(name) }}
-        - tomli    # [py<311]
-        - tomli-w
-    test:
-      imports:
-        - msgspec
+  # Skip s390x as it's optional and fails some tests
+  skip: true  # [py<38 or s390x]
+  script: {{ PYTHON }} -m pip install . --no-deps --no-build-isolation -vv
+      
+requirements:
+  build:
+    - {{ compiler('c') }}
+  host:
+    - python
+    - pip
+    - setuptools
+    - wheel
+  run:
+    - python
+
+test:
+  imports:
+    - msgspec
+  requires:
+    - pip
+    - pytest
+  source_files:
+    - tests
+    - msgspec/_core.c
+    - setup.cfg
+  commands:
+    - pip check
+    - pytest
```

```suggestion
    - pytest tests
```
I'd be explicit about test dir, just in case something gets added in the future to test runtime dir we don't want to run.
Also saves like a 1ms for traversing all the dirs, big deal 😄

---

Good idea ! Thanks :)